File: Go/14-最长公共前缀.go
package main
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	if len(strs) == 1 {
		return strs[0]
	}
	temp := strs[0]
	res := ""
	for i:=0;i<len(temp);i++ {
		for j:=1;j<len(strs);j++ {
			if i == len(strs[j]) || temp[i] != strs[j][i] {
				return  res
			}

		}
		res += string(temp[i]);
	}
	return res
}

File: Python/14-最长公共前缀.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        if len(strs) == 1:
            return strs[0]
        tmin = min(strs)
        tmax = max(strs)
        for i,c in enumerate(tmin):
            if c != tmax[i]:
                return tmin[:i]

        return tmin
